element-connector/src/main/java/org/eclipse/californium/elements/RelaxedCorrelationContextMatcher.java

```diff
@@ -36,29 +36,19 @@ public boolean isResponseRelatedToRequest(CorrelationContext requestContext, Cor
 		return internalMatch(requestContext, responseContext);
 	}
 
+	@Override
+	public boolean isToBeSent(CorrelationContext messageContext, CorrelationContext connectorContext) {
+		return internalMatch(messageContext, connectorContext);
+	}
+
 	private final boolean internalMatch(CorrelationContext requestedContext, CorrelationContext availableContext) {
 		if (null == requestedContext) {
 			return true;
 		} else if (null == availableContext) {
 			return false;
 		}
 		if (requestedContext.get(DtlsCorrelationContext.KEY_SESSION_ID) != null) {
-			boolean match = requestedContext.get(DtlsCorrelationContext.KEY_SESSION_ID).equals(
-					availableContext.get(DtlsCorrelationContext.KEY_SESSION_ID))
-					&& requestedContext.get(DtlsCorrelationContext.KEY_CIPHER).equals(
-							availableContext.get(DtlsCorrelationContext.KEY_CIPHER));
-
-			LOGGER.log(
-					match ? Level.FINEST : Level.WARNING,
-					"(D)TLS session {0}, {1}",
-					new Object[] { requestedContext.get(DtlsCorrelationContext.KEY_SESSION_ID),
-							availableContext.get(DtlsCorrelationContext.KEY_SESSION_ID) });
-			LOGGER.log(
-					match ? Level.FINEST : Level.WARNING,
-					"(D)TLS cipher {0}, {1}",
-					new Object[] { requestedContext.get(DtlsCorrelationContext.KEY_CIPHER),
-							availableContext.get(DtlsCorrelationContext.KEY_CIPHER) });
-			return match;
+			return CorrelationContextUtil.match(getName(), KEYS, requestedContext, availableContext);
```

Comment:
I'm not sure. For me it seems to be more deterministic, if all connectors do their check. But in the end, I think the `CorrelationContext` (and its extendability) should be more specified. Currently having two point of extendability (`CorrelationContext` and `CorrelationContextMatcher`) may lead to issues.